pkg/cluster/manager.go

```diff
@@ -0,0 +1,1886 @@
+		}).
+		Func("DestroyCluster", func(ctx *task.Context) error {
+			return operator.Destroy(ctx, topo, destroyOpt)
+		}).
+		Build()
+
+	if err := t.Execute(task.NewContext()); err != nil {
+		if errorx.Cast(err) != nil {
+			// FIXME: Map possible task errors and give suggestions.
+			return err
+		}
+		return perrs.Trace(err)
+	}
+
+	if err := d.specManager.Remove(clusterName); err != nil {
+		return perrs.Trace(err)
+	}
+
+	log.Infof("Destroyed cluster `%s` successfully", clusterName)
+	return nil
+
+}
+
+// ExecOptions for exec shell command.
+type ExecOptions struct {
+	Command string
+	Sudo    bool
+}
+
+// Exec shell command on host in the tidb cluster.
+func (d *Manager) Exec(clusterName string, opt ExecOptions, gOpt operator.Options) error {
+	metadata, err := d.meta(clusterName)
+	if err != nil {
+		return perrs.AddStack(err)
+	}
+
+	topo := metadata.GetTopology()
+	base := metadata.GetBaseMeta()
+
+	filterRoles := set.NewStringSet(gOpt.Roles...)
+	filterNodes := set.NewStringSet(gOpt.Nodes...)
+
+	var shellTasks []task.Task
+	uniqueHosts := map[string]int{} // host -> ssh-port
+	topo.IterInstance(func(inst spec.Instance) {
+		if _, found := uniqueHosts[inst.GetHost()]; !found {
+			if len(gOpt.Roles) > 0 && !filterRoles.Exist(inst.Role()) {
+				return
+			}
+
+			if len(gOpt.Nodes) > 0 && !filterNodes.Exist(inst.GetHost()) {
+				return
+			}
+
+			uniqueHosts[inst.GetHost()] = inst.GetSSHPort()
+		}
+	})
+
+	for host := range uniqueHosts {
+		shellTasks = append(shellTasks,
+			task.NewBuilder().
+				Shell(host, opt.Command, opt.Sudo).
+				Build())
+	}
+
+	t := task.NewBuilder().
+		SSHKeySet(
+			d.specManager.Path(clusterName, "ssh", "id_rsa"),
+			d.specManager.Path(clusterName, "ssh", "id_rsa.pub")).
+		ClusterSSH(topo, base.User, gOpt.SSHTimeout).
+		Parallel(shellTasks...).
+		Build()
+
+	execCtx := task.NewContext()
+	if err := t.Execute(execCtx); err != nil {
+		if errorx.Cast(err) != nil {
+			// FIXME: Map possible task errors and give suggestions.
+			return err
+		}
+		return perrs.Trace(err)
+	}
+
+	// print outputs
+	for host := range uniqueHosts {
+		stdout, stderr, ok := execCtx.GetOutputs(host)
+		if !ok {
+			continue
+		}
+		log.Infof("Outputs of %s on %s:",
+			color.CyanString(opt.Command),
+			color.CyanString(host))
+		if len(stdout) > 0 {
+			log.Infof("%s:\n%s", color.GreenString("stdout"), stdout)
+		}
+		if len(stderr) > 0 {
+			log.Infof("%s:\n%s", color.RedString("stderr"), stderr)
+		}
+	}
+
+	return nil
+}
+
+// Display cluster meta and topology.
+func (d *Manager) Display(clusterName string, opt operator.Options) error {
+	metadata, err := d.meta(clusterName)
+	if err != nil {
+		return perrs.AddStack(err)
+	}
+
+	topo := metadata.GetTopology()
+	base := metadata.GetBaseMeta()
+
+	// display cluster meta
+	cyan := color.New(color.FgCyan, color.Bold)
+	fmt.Printf("%s Cluster: %s\n", d.sysName, cyan.Sprint(clusterName))
+	fmt.Printf("%s Version: %s\n", d.sysName, cyan.Sprint(base.Version))
```

Comment:
The previous implementation will display the dashboard address, I think we should keep it.

Reply:
it remain the origin logic in ~/code/tiup/components/cluster/command/display.go